akka-bbb-apps/src/main/scala/org/bigbluebutton/core/apps/presentationpod/PresentationUploadTokenReqMsgHdlr.scala

```diff
@@ -68,6 +68,12 @@ trait PresentationUploadTokenReqMsgHdlr extends RightsManagementTrait {
     log.info("handlePresentationUploadTokenReqMsg" + liveMeeting.props.meetingProp.intId +
       " userId=" + msg.header.userId + " filename=" + msg.body.filename)
 
+    if (liveMeeting.props.meetingProp.disabledFeatures.contains("presentation")) {
+      broadcastPresentationUploadTokenFailResp(msg)
+      val meetingId = liveMeeting.props.meetingProp.intId
+      val reason = "Presentation is disabled for this meeting"
+      PermissionCheck.ejectUserForFailedPermission(meetingId, msg.header.userId, reason, bus.outGW, liveMeeting)
+    }
     if (filterPresentationMessage(liveMeeting.users2x, msg.header.userId) &&
```

Comment:
```suggestion
    } else if (filterPresentationMessage(liveMeeting.users2x, msg.header.userId) &&
```

The way you made the condition, the following code would still be executed..